### DISCUSSION TITLE
Trend arrows modification

### DISCUSSION THREAD

**Discussion by manor7777** [NightscoutFoundation/xDrip#3606] [Ideas]
I'm using Xdrip+ as a bridge between apps. I also use a Dexcom device.

When in companion mode, Xdrip reports the same value, but different arrows (e.g ⬆️⬆️ rather than ↗️)

How is it possible to change this to accurately reflect the actual trend? Would it be possible to implement settings to change this? In Dexcom, ⬆️ counts as 0.1-0.3mmol per minute, and 0.3 and above is ⬆️⬆️ (or ⬇️⬇️ you get the idea)

Thanks in advance

**Comment by Navid200** [comment]
If xDrip receives a delta value from its data source, it will boadcast that same delta.
But, if xDrip does not receive a delta from its source, I believe it broadcasts its own delta.  

xDrip also shows its own delta on screen, not what it receives from its data source.

xDrip calculates the delta by subtracting the previous value  from the current value.  It shows that as the current delta or trend arrow.  

At the time an app receives a value that deviates from the current trend, there are two possibilities.  
One is that this is the start of a real deviation.  The other is that this is noise.  
There is no way to know which one it is at that time.  
As far as what an app can do under such circumstances, there are two options.  

1- Assume that it is noise.
Don't reflect the sudden change in the shown delta.  
Wait for the next reading.  If the next reading shows that it indeed is a real deviation, reflect the change in the delta then.  

2- Assume that it is real.
Reflect the sudden change in the shown delta.
If the next reading shows that it was noise, reflect the change back to the then current reading with another delta in the opposite direction.

Each one of those two options has a con.

The con of 1: If it is really a deviation, option 1 delays when you really find out.  So, you receive delayed information.

The con of 2: If it is noise, you may take action to correct something that is not real.  

When I compare those two cons, I prefer 2.  I prefer not to have to put up with any additional delay.  The fact is that what a CGM shows is already delayed by about 10 minutes because interstitial fluid glucose is delayed with respect to blood glucose.  I don't want any more delay.
If I see a sudden change on my CGM, I am the one who decides how to deal with it knowing the facts.  I don't want a piece of code to make that judgement on my behalf.   If I have hypoglycemia, it is me who may have to go to hospital, not the app!

If someone prefers the delayed approach, they can use the official apps.
The following is my personal view.  If a proposal is made to offer a switch in xDrip to let the user decide if xDrip should delay or not, I will strongly object to the proposal.  Adding more and more switches to xDrip will only be a disaster.  There are already too many switches in xDrip.

I am sorry about this long response to a short query.
I am working on a write-up for the guide and this helps me clean up how I am going to write it eventually.

**Reply by manor7777** [reply]
Sorry for the short answer to your essay!

If that's the case, why isn't it receiving the delta in companion mode? Should that be fixed as a bug then?

Since I'm using Xdrip purely as a bridge, and it's not actually directly reading the data from a sensor, rather scraping the notification, it's messing with AndroidAPS's algorithm with more random up and down arrows than it should. It's the same number, it's just not reading the delta!

**Reply by jamorham** [reply]
There can be some improvements made in this area. It has been on my todo list for a while. You may also want to enable the companion bluetooth option.

**Reply by manor7777** [reply]
Edited because low glucose idiot moment

Can you explain what the companion Bluetooth option does? Quick Google didn't find much and the option isn't very descriptive 

Thankyou for your continued support :)

**Reply by Navid200** [reply]
For a Dexcom companion app, it offers backfill.   And may get more info.  It may not work on all devices:
https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html

**Reply by manor7777** [reply]
Unfortunately, turning on this option disrupts (ala disables) the connection to AAPS in my short testing, turning it off immediately restored connection. It unfortunately also still didn't pick up the delta, generating it's own instead, so no real help :( - Through I am using the newer Dexcom one+ app, it didn't work with the G7 app either in my testing (the delta)
